### PR TITLE
Test backup poller with the http2 connector fix

### PR DESCRIPTION
**Pull request by y-zeng**

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                FILE SIZE
 ++++++++++++++ GROWING                                                  ++++++++++++++
  +0.2%    +957 [None]                                                   +16.3Ki  +0.3%
      +0.2%    +857 [Unmapped]                                               +16.3Ki  +0.3%
      +5.5%     +44 [None]                                                         0  [ = ]
      [NEW]     +40 g_poller_mu                                                    0  [ = ]
      [NEW]      +8 g_poller                                                       0  [ = ]
      [NEW]      +4 g_poll_interval_ms                                            +4  [NEW]
      +100%      +4 g_once                                                         0  [ = ]
  [NEW]    +995 src/core/ext/filters/client_channel/backup_poller.cc        +995  [NEW]
      [NEW]    +336 grpc_client_channel_start_backup_polling                    +336  [NEW]
      [NEW]    +222 run_poller                                                  +222  [NEW]
      [NEW]    +211 grpc_client_channel_stop_backup_polling                     +211  [NEW]
      [NEW]    +107 init_globals                                                +107  [NEW]
      [NEW]     +81 backup_poller_shutdown_unref                                 +81  [NEW]
      [NEW]     +36 [Unmapped]                                                   +36  [NEW]
      [NEW]      +2 done_poller                                                   +2  [NEW]
  +0.2%     +32 src/core/ext/filters/client_channel/client_channel.cc        +32  +0.2%
      +1.7%     +16 cc_init_channel_elem                                         +16  +1.7%
      +6.2%     +15 cc_destroy_channel_elem                                      +15  +6.2%
      +0.4%      +1 [Unmapped]                                                    +1  +0.4%
  +1.0%     +16 src/core/ext/transport/chttp2/client/chttp2_connector.cc     +16  +1.0%
      +3.2%     +16 connected                                                    +16  +3.2%

  +0.2% +1.95Ki TOTAL                                                    +17.3Ki  +0.3%


****************************************************************

libgrpc++.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++

 -------------- SHRINKING                                                                            --------------
  -0.9% -1.95Ki [None]                                                                               -38.7Ki  -0.8%
      -0.9% -1.74Ki [Unmapped]                                                                           -38.5Ki  -0.8%
      [DEL]     -51 typeinfo name for grpc::(anonymous namespace)::ChannelConnectivityWatcher                -51  [DEL]
     -23.5%     -40 [None]                                                                                     0  [ = ]
      [DEL]     -40 grpc::(anonymous namespace)::ChannelConnectivityWatcher::g_watcher_mu_                     0  [ = ]
      [DEL]     -40 typeinfo for grpc::(anonymous namespace)::ChannelConnectivityWatcher                     -40  [DEL]
      [DEL]     -32 vtable for grpc::(anonymous namespace)::ChannelConnectivityWatcher                       -32  [DEL]
      [DEL]      -8 grpc::(anonymous namespace)::ChannelConnectivityWatcher::g_watcher_                        0  [ = ]
 -44.2% -3.06Ki src/cpp/client/channel_cc.cc                                                         -3.06Ki -44.2%
     -80.0% -1.01Ki grpc::Channel::Channel                                                               -1.01Ki -80.0%
      [DEL]    -765 grpc::(anonymous namespace)::WatchStateChange                                           -765  [DEL]
     -72.0%    -657 grpc::Channel::~Channel                                                                 -657 -72.0%
      [DEL]    -333 grpc::(anonymous namespace)::ChannelConnectivityWatcher::~ChannelConnectivityWatcher    -333  [DEL]
      [DEL]    -332 grpc::(anonymous namespace)::ChannelConnectivityWatcher::~ChannelConnectivityWatcher    -332  [DEL]
      [DEL]    -194 void std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >    -194  [DEL]
      -7.7%     -15 [Unmapped]                                                                               -15  -7.7%

  -1.5% -5.01Ki TOTAL                                                                                -41.7Ki  -0.8%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                             FILE SIZE
 ++++++++++++++ GROWING                                               ++++++++++++++
  +0.2%   +1005 [None]                                                +16.3Ki  +0.3%
      +0.2%    +905 [Unmapped]                                            +16.3Ki  +0.3%
      +5.5%     +44 [None]                                                      0  [ = ]
      [NEW]     +40 g_poller_mu                                                 0  [ = ]
      [NEW]      +8 g_poller                                                    0  [ = ]
      [NEW]      +4 g_poll_interval_ms                                         +4  [NEW]
      +100%      +4 g_once                                                      0  [ = ]
  [NEW]    +995 src/core/ext/filters/client_channel/backup_poller.cc     +995  [NEW]
      [NEW]    +336 grpc_client_channel_start_backup_polling                 +336  [NEW]
      [NEW]    +222 run_poller                                               +222  [NEW]
      [NEW]    +211 grpc_client_channel_stop_backup_polling                  +211  [NEW]
      [NEW]    +107 init_globals                                             +107  [NEW]
      [NEW]     +81 backup_poller_shutdown_unref                              +81  [NEW]
      [NEW]     +36 [Unmapped]                                                +36  [NEW]
      [NEW]      +2 done_poller                                                +2  [NEW]
  +0.2%     +32 src/core/ext/filters/client_channel/client_channel.cc     +32  +0.2%
      +1.7%     +16 cc_init_channel_elem                                      +16  +1.7%
      +6.2%     +15 cc_destroy_channel_elem                                   +15  +6.2%
      +0.4%      +1 [Unmapped]                                                 +1  +0.4%

  +0.2% +1.98Ki TOTAL                                                 +17.3Ki  +0.3%


****************************************************************

libgrpc++.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++

 -------------- SHRINKING                                                                            --------------
  -0.9% -1.95Ki [None]                                                                               -38.7Ki  -0.8%
      -0.9% -1.74Ki [Unmapped]                                                                           -38.5Ki  -0.8%
      [DEL]     -51 typeinfo name for grpc::(anonymous namespace)::ChannelConnectivityWatcher                -51  [DEL]
     -23.5%     -40 [None]                                                                                     0  [ = ]
      [DEL]     -40 grpc::(anonymous namespace)::ChannelConnectivityWatcher::g_watcher_mu_                     0  [ = ]
      [DEL]     -40 typeinfo for grpc::(anonymous namespace)::ChannelConnectivityWatcher                     -40  [DEL]
      [DEL]     -32 vtable for grpc::(anonymous namespace)::ChannelConnectivityWatcher                       -32  [DEL]
      [DEL]      -8 grpc::(anonymous namespace)::ChannelConnectivityWatcher::g_watcher_                        0  [ = ]
 -44.2% -3.06Ki src/cpp/client/channel_cc.cc                                                         -3.06Ki -44.2%
     -80.0% -1.01Ki grpc::Channel::Channel                                                               -1.01Ki -80.0%
      [DEL]    -765 grpc::(anonymous namespace)::WatchStateChange                                           -765  [DEL]
     -72.0%    -657 grpc::Channel::~Channel                                                                 -657 -72.0%
      [DEL]    -333 grpc::(anonymous namespace)::ChannelConnectivityWatcher::~ChannelConnectivityWatcher    -333  [DEL]
      [DEL]    -332 grpc::(anonymous namespace)::ChannelConnectivityWatcher::~ChannelConnectivityWatcher    -332  [DEL]
      [DEL]    -194 void std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >    -194  [DEL]
      -7.7%     -15 [Unmapped]                                                                               -15  -7.7%

  -1.5% -5.01Ki TOTAL                                                                                -41.7Ki  -0.8%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: src/core/ext/filters/client_channel/backup_poller.cc [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+static void g_poller_unref(grpc_exec_ctx* exec_ctx) {
+  if (gpr_unref(&g_poller->refs)) {
+    gpr_mu_lock(&g_poller_mu);
+    backup_poller* p = g_poller;
+    g_poller = NULL;
+    gpr_mu_unlock(&g_poller_mu);
+    gpr_mu_lock(p->pollset_mu);
+    p->shutting_down = true;
+    grpc_pollset_shutdown(exec_ctx, p->pollset,
```
Comment (dgquintas) [inline src/core/ext/filters/client_channel/backup_poller.cc]: can't we save `p->pollset` in a local variable and unlock `p->pollset_mu` before calling `grpc_pollset_shutdown`? We'd also need to init the shutdown closure inside the lock I believe.

Not sure we care about performance on this path though.

File: src/core/ext/filters/client_channel/backup_poller.cc [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+void grpc_client_channel_start_backup_polling(
+    grpc_exec_ctx* exec_ctx, grpc_pollset_set* interested_parties) {
+  gpr_once_init(&g_once, init_globals);
+  if (g_poll_interval_ms == 0) {
+    return;
+  }
+  gpr_mu_lock(&g_poller_mu);
+  if (g_poller == NULL) {
+    g_poller = (backup_poller*)gpr_zalloc(sizeof(backup_poller));
```
Comment (dgquintas) [inline src/core/ext/filters/client_channel/backup_poller.cc]: nit: I'd put the body of this `if` inside its own function (something like `g_poller_init()`), for readability.

File: src/core/ext/filters/client_channel/backup_poller.h [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/*
+ *
+ * Copyright 2015 gRPC authors.
```
Comment (dgquintas) [inline src/core/ext/filters/client_channel/backup_poller.h]: s/2015/2017

File: src/core/ext/filters/client_channel/backup_poller.cc (hunk 1 above)
Comment (y-zeng) [reply src/core/ext/filters/client_channel/backup_poller.cc]: `p->pollset` is required to be held while calling `grpc_pollset_shutdown`. This should not be in the performance critical path. It's part of the channel shutdown process.

**Pull request closed without merging**